pkg/inspector/image-inspector.go

```diff
@@ -69,28 +69,32 @@ func (i *defaultImageInspector) Inspect() error {
 		return fmt.Errorf("Unable to connect to docker daemon: %v\n", err)
 	}
 
-	if _, err := client.InspectImage(i.opts.Image); err != nil {
-		log.Printf("Pulling image %s", i.opts.Image)
-		imagePullOption := docker.PullImageOptions{Repository: i.opts.Image}
-
-		var imagePullAuths *docker.AuthConfigurations
-		var authCfgErr error
-		if imagePullAuths, authCfgErr = i.getAuthConfigs(); authCfgErr != nil {
-			return authCfgErr
-		}
+	imageMetaBefore, inspectErrBefore := client.InspectImage(i.opts.Image)
 
-		// Try all the possible auth's from the config file
-		var authErr error
-		for _, auth := range imagePullAuths.Configs {
-			if authErr = client.PullImage(imagePullOption, auth); authErr == nil {
-				break
-			}
-		}
-		if authErr != nil {
-			return fmt.Errorf("Unable to pull docker image: %v\n", authErr)
+	log.Printf("Pulling image %s", i.opts.Image)
+	imagePullOption := docker.PullImageOptions{Repository: i.opts.Image}
+
+	var imagePullAuths *docker.AuthConfigurations
+	var authCfgErr error
+	if imagePullAuths, authCfgErr = i.getAuthConfigs(); authCfgErr != nil {
+		return authCfgErr
+	}
+
+	// Try all the possible auth's from the config file
+	var authErr error
+	for _, auth := range imagePullAuths.Configs {
+		if authErr = client.PullImage(imagePullOption, auth); authErr == nil {
```

Comment:
not relevant to this change but I noticed that we only ever give the last auth error assuming all `PullImage` calls fail.  Maybe we should be aggregating them and showing them all.